generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -81,27 +73,19 @@ private boolean mustBeNull(FieldSpec fieldSpec) {
         return Collections.singletonList(new CannedValuesFieldValueSource(whitelist.collect(Collectors.toList())));
     }
 
-    private Stream<FieldSpec> getNotNullSetRestrictionFilterOnMustContainRestriction(MustContainRestriction restriction) {
-        return restriction.getRequiredObjects()
-            .stream()
-            .filter(o -> o.getSetRestrictions() != null
-                && o.getSetRestrictions().getWhitelist() != null);
-    }
-
     private boolean mayBeNull(FieldSpec fieldSpec) {
         return fieldSpec.getNullRestrictions() == null;
     }
 
     private List<FieldValueSource> getMustContainRestrictionSources(FieldSpec fieldSpec) {
         Set<FieldSpec> mustContainRestrictionFieldSpecs = fieldSpec.getMustContainRestriction().getRequiredObjects();
-        if (mustContainRestrictionFieldSpecs.size() > 1) {
-            mustContainRestrictionFieldSpecs = mustContainRestrictionReducer.getReducedMustContainRestriction(fieldSpec);
-        }
+
+        FilterAfterFirst<FieldValueSource> nullFilter = new FilterAfterFirst(nullOnlySource);
```

Comment:
See [this comment](https://github.com/ScottLogic/datahelix/pull/679/files#r264563059)